x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -130,48 +130,60 @@ export function getPreviewTransformRequestBody(
   return request;
 }
 
+export const getCreateTransformSettingsRequestBody = (
+  transformDetailsState: StepDetailsExposedState
+): { settings: PutTransformsRequestSchema['settings'] } => {
+  // have to use Record<> here because typeof schema is readonly
+  const tempSettings = {} as Record<string, number | null | undefined>;
+
+  if (transformDetailsState.transformSettingsMaxPageSearchSize) {
+    tempSettings.max_page_search_size = transformDetailsState.transformSettingsMaxPageSearchSize;
+  }
+  if (transformDetailsState.transformSettingsDocsPerSecond) {
+    tempSettings.docs_per_second = transformDetailsState.transformSettingsDocsPerSecond;
+  }
+
+  return { settings: tempSettings };
+};
+
 export const getCreateTransformRequestBody = (
   indexPatternTitle: IndexPattern['title'],
   pivotState: StepDefineExposedState,
   transformDetailsState: StepDetailsExposedState
-): PutTransformsRequestSchema => ({
-  ...getPreviewTransformRequestBody(
-    indexPatternTitle,
-    getPivotQuery(pivotState.searchQuery),
-    dictionaryToArray(pivotState.groupByList),
-    dictionaryToArray(pivotState.aggList)
-  ),
-  // conditionally add optional description
-  ...(transformDetailsState.transformDescription !== ''
-    ? { description: transformDetailsState.transformDescription }
-    : {}),
-  // conditionally add optional frequency
-  ...(transformDetailsState.transformFrequency !== ''
-    ? { frequency: transformDetailsState.transformFrequency }
-    : {}),
-  dest: {
-    index: transformDetailsState.destinationIndex,
-  },
-  // conditionally add continuous mode config
-  ...(transformDetailsState.isContinuousModeEnabled
-    ? {
-        sync: {
-          time: {
-            field: transformDetailsState.continuousModeDateField,
-            delay: transformDetailsState.continuousModeDelay,
+): PutTransformsRequestSchema => {
+  return {
```

Comment:
With the `settings` part moved to its own function (`getCreateTransformSettingsRequestBody()`), you should now be able to revert this change with the additional `return` again to the previous code using just `({ ... })`.

Reply:
Updated here https://github.com/elastic/kibana/pull/78364/commits/665232fd43a76aa120784f80400a2878f383a08b